Fixes build error:

  <flex>/lib/libfl.so: undefined reference to `yylex'

The same patch exists in upstream thrift git repo, commit
ec8daae71004b3c2346bf12b2d74e52ed0815337
"THRIFT-2386: Thrift refuses to link yylex".
diff --git a/compiler/cpp/src/main.h b/compiler/cpp/src/main.h
index 87af5f6..27285f5 100644
--- a/compiler/cpp/src/main.h
+++ b/compiler/cpp/src/main.h
@@ -28,8 +28,9 @@
 /**
  * Defined in the flex library
  */
-
-int yylex(void);
+extern "C" {
+	int yylex(void);
+}
 
 int yyparse(void);
 
